src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -131,6 +142,42 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
       `
     );
   }
+  /**
+   * Allows developer to define array of MicrosoftGraph.Person for selectedPeople
+   *
+   * @param {[MicrosoftGraph.UserIdentity]} [id]
+   * @returns {[MicrosoftGraph.Person]}
+   * @memberof MgtPeoplePicker
+   */
+  public async selectUsersById(people) {
+    const provider = Providers.globalProvider;
+    if (provider && provider.state === ProviderState.SignedIn) {
+      const client = Providers.globalProvider.graph;
+      const peopleDetails = [];
+      const name = [];
+      const peopleArr = [];
+      if (people) {
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < people.length; i++) {
+          // first find valid UserName
+          peopleDetails.push(await client.getUser(people[i]));
+        }
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < peopleDetails.length; i++) {
+          // get people details to push
+          name.push(peopleDetails[i].displayName);
+        }
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < name.length; i++) {
+          peopleArr.push(await client.findPerson(name[i]));
+        }
+        // tslint:disable-next-line: prefer-for-of
+        for (let i = 0; i < peopleArr.length; i++) {
+          this.addPerson(peopleArr[i][0]);
```

Comment:
the `addPerson` function does more than necessary. I think it's enough to just push to the selectedPeople and raise the event only once at the end of this function

Reply:
the `this.people = [];` logic had to be kept in to update the list (it also acts to update the selected people ui). Removed the unnecessary features